### PR TITLE
cleanup:e2e:stop using deprecated framework.ExpectNotEqual

### PR DESCRIPTION
**Pull request by shijinye**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
cleanup:e2e:stop using deprecated framework.ExpectNotEqual

#### Which issue(s) this PR fixes:

Fixes https://github.com/kubernetes/kubernetes/pull/115961

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
NONE
```

### CODE REVIEW

File: test/e2e/common/node/pods.go [hunk 1]
```diff
@@ -328,7 +328,7 @@ var _ = SIGDescribe("Pods", func() {
 		}
 
 		gomega.Expect(lastPod.DeletionTimestamp).ToNot(gomega.BeNil())
-		gomega.Expect(lastPod.Spec.TerminationGracePeriodSeconds).ToNot(gomega.BeZero())
```
Comment (kerthcet) [inline test/e2e/common/node/pods.go:331]: Why change this? `BeZero` seems more simple?

Comment (carlory) [reply test/e2e/common/node/pods.go:331]: I think `gomega.BeNil()` is expected.

Comment (shijinye) [reply test/e2e/common/node/pods.go]: adjusted to Bezero

**Review by kerthcet — COMMENTED** [review]
/lgtm
I have no approval privilege  for e2e package.

File: test/e2e_node/pod_conditions_test.go [hunk 2]
```diff
@@ -125,7 +126,7 @@ func runPodFailingConditionsTest(f *framework.Framework, hasInitContainers, chec
 			// Verify PodInitialized is set if init containers are not present (since without init containers, it gets set very early)
 			initializedTime, err := getTransitionTimeForPodConditionWithStatus(p, v1.PodInitialized, true)
 			framework.ExpectNoError(err)
-			framework.ExpectNotEqual(initializedTime.Before(scheduledTime), true, fmt.Sprintf("pod without init containers is initialized at: %v which is before pod scheduled at: %v", initializedTime, scheduledTime))
+			gomega.Expect(initializedTime.Before(scheduledTime)).ToNot(gomega.BeTrue(), fmt.Sprintf("pod without init containers is initialized at: %v which is before pod scheduled at: %v", initializedTime, scheduledTime))
```
Comment (carlory) [inline test/e2e_node/pod_conditions_test.go:129]: Can you change this to not use BeTrue? See https://github.com/kubernetes/kubernetes/issues/105678.

FYI: https://github.com/kubernetes/kubernetes/pull/119558/files#diff-ca60431bd6cf2bd182b588d9b8c9135be21d1a0d49c7c2a8b60826620b4c83cfR151

File: test/e2e_node/pod_conditions_test.go [hunk 3]
```diff
@@ -165,28 +166,28 @@ func runPodReadyConditionsTest(f *framework.Framework, hasInitContainers, checkP
 
 			if hasInitContainers {
 				// With init containers, verify the sequence of conditions is: Scheduled => PodReadyToStartContainers => Initialized
-				framework.ExpectNotEqual(readyToStartContainersTime.Before(scheduledTime), true, fmt.Sprintf("pod with init containers is initialized at: %v which is before pod has ready to start at: %v", initializedTime, readyToStartContainersTime))
-				framework.ExpectNotEqual(initializedTime.Before(readyToStartContainersTime), true, fmt.Sprintf("pod with init containers is initialized at: %v which is before pod has ready to start at: %v", initializedTime, readyToStartContainersTime))
+				gomega.Expect(readyToStartContainersTime.Before(scheduledTime)).ToNot(gomega.BeTrue(), fmt.Sprintf("pod with init containers is initialized at: %v which is before pod has ready to start at: %v", initializedTime, readyToStartContainersTime))
```
Comment (carlory) [inline test/e2e_node/pod_conditions_test.go:169]: and here

File: test/e2e_node/pod_conditions_test.go [hunk 4]
```diff
@@ -165,28 +166,28 @@ func runPodReadyConditionsTest(f *framework.Framework, hasInitContainers, checkP
...
+				gomega.Expect(initializedTime.Before(readyToStartContainersTime)).ToNot(gomega.BeTrue(), fmt.Sprintf("pod with init containers is initialized at: %v which is before pod has ready to start at: %v", initializedTime, readyToStartContainersTime))
```
Comment (carlory) [inline test/e2e_node/pod_conditions_test.go:170]: and here.

File: test/e2e_node/pod_conditions_test.go [hunk 5]
```diff
@@ -165,28 +166,28 @@ func runPodReadyConditionsTest(f *framework.Framework, hasInitContainers, checkP
...
 			} else {
 				// Without init containers, verify the sequence of conditions is: Scheduled => Initialized => PodReadyToStartContainers
 				condBeforeContainersReadyTransitionTime = readyToStartContainersTime
 				errSubstrIfContainersReadyTooEarly = "ready to start"
-				framework.ExpectNotEqual(initializedTime.Before(scheduledTime), true, fmt.Sprintf("pod without init containers initialized at: %v which is before pod scheduled at: %v", initializedTime, scheduledTime))
-				framework.ExpectNotEqual(readyToStartContainersTime.Before(initializedTime), true, fmt.Sprintf("pod without init containers has ready to start at: %v which is before pod is initialized at: %v", readyToStartContainersTime, initializedTime))
+				gomega.Expect(initializedTime.Before(scheduledTime)).ToNot(gomega.BeTrue(), fmt.Sprintf("pod without init containers initialized at: %v which is before pod scheduled at: %v", initializedTime, scheduledTime))
```
Comment (carlory) [inline test/e2e_node/pod_conditions_test.go:175]: and here.

File: test/e2e_node/pod_conditions_test.go [hunk 6]
```diff
@@ -165,28 +166,28 @@ func runPodReadyConditionsTest(f *framework.Framework, hasInitContainers, checkP
...
+				gomega.Expect(readyToStartContainersTime.Before(initializedTime)).ToNot(gomega.BeTrue(), fmt.Sprintf("pod without init containers has ready to start at: %v which is before pod is initialized at: %v", readyToStartContainersTime, initializedTime))
```
Comment (carlory) [inline test/e2e_node/pod_conditions_test.go:176]: and here

File: test/e2e_node/pod_conditions_test.go [hunk 7]
```diff
@@ -165,28 +166,28 @@ func runPodReadyConditionsTest(f *framework.Framework, hasInitContainers, checkP
...
 			}
 		} else {
 			// In the absence of PodHasReadyToStartContainers feature disabled, verify the sequence is: Scheduled => Initialized
-			framework.ExpectNotEqual(initializedTime.Before(scheduledTime), true, fmt.Sprintf("pod initialized at: %v which is before pod scheduled at: %v", initializedTime, scheduledTime))
+			gomega.Expect(initializedTime.Before(scheduledTime)).ToNot(gomega.BeTrue(), fmt.Sprintf("pod initialized at: %v which is before pod scheduled at: %v", initializedTime, scheduledTime))
```
Comment (carlory) [inline test/e2e_node/pod_conditions_test.go:180]: and here.

File: test/e2e_node/pod_conditions_test.go [hunk 8]
```diff
@@ -165,28 +166,28 @@ func runPodReadyConditionsTest(f *framework.Framework, hasInitContainers, checkP
...
 		}
 		// Verify the next condition to get set is ContainersReady
 		containersReadyTime, err := getTransitionTimeForPodConditionWithStatus(p, v1.ContainersReady, true)
 		framework.ExpectNoError(err)
-		framework.ExpectNotEqual(containersReadyTime.Before(condBeforeContainersReadyTransitionTime), true, fmt.Sprintf("containers ready at: %v which is before pod %s: %v", containersReadyTime, errSubstrIfContainersReadyTooEarly, initializedTime))
+		gomega.Expect(containersReadyTime.Before(condBeforeContainersReadyTransitionTime)).ToNot(gomega.BeTrue(), fmt.Sprintf("containers ready at: %v which is before pod %s: %v", containersReadyTime, errSubstrIfContainersReadyTooEarly, initializedTime))
```
Comment (carlory) [inline test/e2e_node/pod_conditions_test.go:185]: and here.

File: test/e2e_node/pod_conditions_test.go [hunk 9]
```diff
@@ -165,28 +166,28 @@ func runPodReadyConditionsTest(f *framework.Framework, hasInitContainers, checkP
...
 
 		// Verify ContainersReady => PodReady
 		podReadyTime, err := getTransitionTimeForPodConditionWithStatus(p, v1.PodReady, true)
 		framework.ExpectNoError(err)
-		framework.ExpectNotEqual(podReadyTime.Before(containersReadyTime), true, fmt.Sprintf("pod ready at: %v which is before pod containers ready at: %v", podReadyTime, containersReadyTime))
+		gomega.Expect(podReadyTime.Before(containersReadyTime)).ToNot(gomega.BeTrue(), fmt.Sprintf("pod ready at: %v which is before pod containers ready at: %v", podReadyTime, containersReadyTime))
```
Comment (carlory) [inline test/e2e_node/pod_conditions_test.go:190]: and here.

**Review by carlory — CHANGES_REQUESTED** [review]

File: test/e2e_node/pod_conditions_test.go (hunk 2 above)
Comment (shijinye) [reply test/e2e_node/pod_conditions_test.go]: Resolved

File: test/e2e_node/pod_conditions_test.go (hunk 5 above)
Comment (shijinye) [reply test/e2e_node/pod_conditions_test.go]: Resolved

File: test/e2e_node/pod_conditions_test.go (hunk 6 above)
Comment (shijinye) [reply test/e2e_node/pod_conditions_test.go]: Resolved

File: test/e2e_node/pod_conditions_test.go (hunk 7 above)
Comment (shijinye) [reply test/e2e_node/pod_conditions_test.go]: Resolved

File: test/e2e_node/pod_conditions_test.go (hunk 8 above)
Comment (shijinye) [reply test/e2e_node/pod_conditions_test.go]: Resolved

File: test/e2e_node/pod_conditions_test.go (hunk 9 above)
Comment (shijinye) [reply test/e2e_node/pod_conditions_test.go]: Resolved

File: test/e2e_node/pod_conditions_test.go (hunk 3 above)
Comment (shijinye) [reply test/e2e_node/pod_conditions_test.go]: Resolved

File: test/e2e_node/eviction_test.go [hunk 10]
```diff
@@ -711,8 +711,7 @@ func verifyEvictionOrdering(ctx context.Context, f *framework.Framework, testSpe
 			}
 		}
 		gomega.Expect(priorityPod).NotTo(gomega.BeNil())
-		framework.ExpectNotEqual(priorityPod.Status.Phase, v1.PodSucceeded,
-			fmt.Sprintf("pod: %s succeeded unexpectedly", priorityPod.Name))
+		gomega.Expect(priorityPod.Status.Phase).ToNot(gomega.Equal(v1.PodSucceeded), fmt.Sprintf("pod: %s succeeded unexpectedly", priorityPod.Name))
```
Comment (carlory) [inline test/e2e_node/eviction_test.go:714]:
```suggestion
		gomega.Expect(priorityPod.Status.Phase).ToNot(gomega.Equal(v1.PodSucceeded), "pod: %s succeeded unexpectedly", priorityPod.Name)
```

File: test/e2e_node/eviction_test.go [hunk 11]
```diff
@@ -726,9 +725,8 @@ func verifyEvictionOrdering(ctx context.Context, f *framework.Framework, testSpe
 			}
 			gomega.Expect(lowPriorityPod).NotTo(gomega.BeNil())
 			if priorityPodSpec.evictionPriority < lowPriorityPodSpec.evictionPriority && lowPriorityPod.Status.Phase == v1.PodRunning {
-				framework.ExpectNotEqual(priorityPod.Status.Phase, v1.PodFailed,
-					fmt.Sprintf("priority %d pod: %s failed before priority %d pod: %s",
-						priorityPodSpec.evictionPriority, priorityPodSpec.pod.Name, lowPriorityPodSpec.evictionPriority, lowPriorityPodSpec.pod.Name))
+				gomega.Expect(priorityPod.Status.Phase).ToNot(gomega.Equal(v1.PodFailed), fmt.Sprintf("priority %d pod: %s failed before priority %d pod: %s",
```
Comment (carlory) [inline test/e2e_node/eviction_test.go:728]: and here.

File: test/e2e_node/eviction_test.go (hunk 10 above)
Comment (shijinye) [reply test/e2e_node/eviction_test.go]: Resolved

File: test/e2e_node/eviction_test.go [hunk 12]
```diff
@@ -739,8 +737,7 @@ func verifyEvictionOrdering(ctx context.Context, f *framework.Framework, testSpe
 
 		// EvictionPriority 0 pods should not fail
 		if priorityPodSpec.evictionPriority == 0 {
-			framework.ExpectNotEqual(priorityPod.Status.Phase, v1.PodFailed,
-				fmt.Sprintf("priority 0 pod: %s failed", priorityPod.Name))
+			gomega.Expect(priorityPod.Status.Phase).ToNot(gomega.Equal(v1.PodFailed), fmt.Sprintf("priority 0 pod: %s failed", priorityPod.Name))
```
[the remark this answers is not in the document]
Comment (shijinye) [reply test/e2e_node/eviction_test.go]: Resolved

File: test/e2e_node/pod_conditions_test.go [hunk 13]
```diff
@@ -125,7 +125,9 @@ func runPodFailingConditionsTest(f *framework.Framework, hasInitContainers, chec
 			// Verify PodInitialized is set if init containers are not present (since without init containers, it gets set very early)
 			initializedTime, err := getTransitionTimeForPodConditionWithStatus(p, v1.PodInitialized, true)
 			framework.ExpectNoError(err)
-			framework.ExpectNotEqual(initializedTime.Before(scheduledTime), true, fmt.Sprintf("pod without init containers is initialized at: %v which is before pod scheduled at: %v", initializedTime, scheduledTime))
+			if initializedTime.Before(scheduledTime) {
```
Comment (tzneal) [inline test/e2e_node/pod_conditions_test.go:128]: Any reason not to:

```go
gomega.Expect(initializedTime.Before(scheduledTime)).To(gomega.BeFalse(),"pod without init containers is initialized at: %v which is before pod scheduled at: %v", initializedTime, scheduledTime)
```

File: test/e2e_node/pod_conditions_test.go [hunk 14]
```diff
@@ -165,28 +167,42 @@ func runPodReadyConditionsTest(f *framework.Framework, hasInitContainers, checkP
 
 			if hasInitContainers {
 				// With init containers, verify the sequence of conditions is: Scheduled => PodReadyToStartContainers => Initialized
-				framework.ExpectNotEqual(readyToStartContainersTime.Before(scheduledTime), true, fmt.Sprintf("pod with init containers is initialized at: %v which is before pod has ready to start at: %v", initializedTime, readyToStartContainersTime))
-				framework.ExpectNotEqual(initializedTime.Before(readyToStartContainersTime), true, fmt.Sprintf("pod with init containers is initialized at: %v which is before pod has ready to start at: %v", initializedTime, readyToStartContainersTime))
+				if readyToStartContainersTime.Before(scheduledTime) {
```
Comment (tzneal) [inline test/e2e_node/pod_conditions_test.go:170]: Same as above

File: test/e2e_node/pod_conditions_test.go (hunk 13 above)
Comment (tzneal) [reply test/e2e_node/pod_conditions_test.go:128]: Thanks, I hadn't seen that.  Not 100% in agreement, but SGTM.

**Review by tzneal — APPROVED** [review]
/lgtm

**Review by SergeyKanzhelev — APPROVED** [review]
/lgtm
/approve

**Pull request merged**